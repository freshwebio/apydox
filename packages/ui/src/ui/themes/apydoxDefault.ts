import { ApydoxTheme } from './types'

const fontFamilies = {
  default: 'Lato, sans-serif',
  headings: 'Roboto Slab, serif',
}

const reusedColours = {
  onPrimary: '#1c1b25',
  onPrimaryDark: '#ffffff',
  onSecondaryLight: '#0f0f13',
}

const breakpoints = ['319px', '599px']

const apydoxDefault: ApydoxTheme = {
  name: 'apydox default theme',
  // Old names, keep around while transitioning.
  backgroundColour: '#1e1e1e',
  foregroundColour: '#fefefe',
  backgroundColourLight: '#414141',
  backgroundColourLighter: '#5b5b5b',
  // New theme structure.
  colours: {
    ...reusedColours,
    primary: '#60affe',
    primaryDark: '#3e70a3',
    primaryLight: '#91c8ff',
    onPrimaryLight: reusedColours.onPrimary,
    secondary: '#4e4d64',
    onSecondary: reusedColours.onPrimaryDark,
    secondaryLight: '#a9a7d9',
    onSecondaryLight: reusedColours.onSecondaryLight,
    tertiary: '#3d39cc',
    onTertiary: reusedColours.onPrimaryDark,
    tertiaryLight: '#a19ff5',
    onTertiaryLight: reusedColours.onSecondaryLight,
    tertiaryDark: '#292782',
    onTertiaryDark: reusedColours.onPrimaryDark,
    quaternary: '#ffd0a1',
    onQuaternary: '#3b3a4c',
    quinary: '#8d4016',
    onQuinary: reusedColours.onPrimaryDark,
    greyScale: {
      white: '#fefefe',
      onWhite: reusedColours.onPrimary,
      grey: '#c4c4c4',
      onGrey: '#131315',
      black: '#1f2327',
      onBlack: reusedColours.onPrimaryDark,
    },
    indicitave: {
      green: '#1f573d',
      onGreen: reusedColours.onPrimaryDark,
      red: '#8f2424',
      onRed: reusedColours.onPrimaryDark,
    },
  },
  textStyles: {
    canon: {
      as: 'h1',
      fontSize: [11, 12, 14],
      lineHeight: ['32px', '36px', '48px'],
      fontFamily: fontFamilies.headings,
    },
    trafalgar: {
      as: 'h1',
      fontSize: [6, 9, 12],
      lineHeight: ['24px', '28px', '36px'],
      fontFamily: fontFamilies.headings,
    },
    paragon: {
      as: 'h2',
      fontSize: [6, 8, 11],
      fontWeight: 500,
      fontFamily: fontFamilies.headings,
      lineHeight: ['24px', '26px', '32px'],
    },
    doublePica: {
      fontSize: [6, 6, 9],
      fontWeight: 400,
      fontFamily: fontFamilies.default,
      lineHeight: ['24px', '24px', '28px'],
    },
    greatPrimer: {
      as: 'h3',
      fontWeight: 400,
      fontSize: [5, 5, 6],
      fontFamily: fontFamilies.headings,
      lineHeight: ['22px', '22px', '24px'],
    },
    bodyCopy: {
      as: 'p',
      fontSize: [3, 4, 4],
      fontFamily: fontFamilies.default,
      lineHeight: ['20px', '22px', '22px'],
    },
    pica: {
      fontSize: [3, 4, 4],
      fontFamily: fontFamilies.default,
      lineHeight: ['20px', '20px', '20px'],
    },
    longPrimer: {
      fontSize: [3, 3, 2],
      fontFamily: fontFamilies.default,
      lineHeight: ['18px', '18px', '18px'],
    },
    brevier: {
      fontSize: [2, 2, 1],
      fontFamily: fontFamilies.default,
      lineHeight: ['16px', '18px', '16px'],
    },
    minion: {
      fontSize: [0, 0, 0],
      fontFamily: fontFamilies.default,
      lineHeight: ['16px', '16px', '16px'],
      textTransform: 'uppercase',
    },
  },
  extraTextStyles: {
    canon: {
      fontSize: { mdTouch: 15 },
      lineHeight: { mdTouch: '56px' },
    },
    trafalgar: {
      fontSize: { mdTouch: 13 },
      lineHeight: { mdTouch: '40px' },
    },
    paragon: {
      fontSize: { mdTouch: 11 },
      lineHeight: { mdTouch: '32px' },
    },
    doublePica: {
      fontSize: { mdTouch: 10 },
      lineHeight: { mdTouch: '30px' },
    },
    greatPrimer: {
      fontSize: { mdTouch: 7 },
      lineHeight: { mdTouch: '24px' },
    },
    bodyCopy: {
      fontSize: { mdTouch: 5 },
      lineHeight: { mdTouch: '24px' },
    },
    pica: {
      fontSize: { mdTouch: 5 },
      lineHeight: { mdTouch: '22px' },
    },
    longPrimer: {
      fontSize: { mdTouch: 3 },
      lineHeight: { mdTouch: '20px' },
    },
    brevier: {
      fontSize: { mdTouch: 2 },
      lineHeight: { mdTouch: '18px' },
    },
    minion: {
      fontSize: { mdTouch: 1 },
      lineHeight: { mdTouch: '16px' },
    },
  },
  fontSizes: [12, 13, 14, 15, 16, 18, 20, 21, 22, 24, 26, 28, 32, 36, 44, 52],
  breakpoints,
  mediaQueries: {
    sm: `@media screen and (min-width: ${breakpoints[0]})`,
    mdTouch: `@media screen and (min-width: ${breakpoints[1]})`,
    md: `@media screen and (min-width: ${breakpoints[1]})`,
  },
}

export default apydoxDefault
