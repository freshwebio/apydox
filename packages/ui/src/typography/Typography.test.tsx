import React from 'react'
import { create } from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import apydoxDefault from '@ui/themes/apydoxDefault'
import {
  Canon,
  Trafalgar,
  Paragon,
  DoublePica,
  GreatPrimer,
  BodyCopy,
  Pica,
  PicaSerif,
  LongPrimer,
  Brevier,
  Minion,
} from './Typography'

const testRenders = (reactElem: React.ReactElement): void => {
  it('should render without any issues', () => {
    const tree = create(
      <ThemeProvider theme={apydoxDefault}>{reactElem}</ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
}

describe('Typography', () => {
  describe('Canon', () => {
    testRenders(<Canon>This is some canon text</Canon>)
  })

  describe('Trafalgar', () => {
    testRenders(<Trafalgar>This is some trafalgar text</Trafalgar>)
  })

  describe('Paragon', () => {
    testRenders(<Paragon>This is some paragon text</Paragon>)
  })

  describe('DoublePica', () => {
    testRenders(<DoublePica>This is some double pica text</DoublePica>)
  })

  describe('GreatPrimer', () => {
    testRenders(<GreatPrimer>This is some great primer text</GreatPrimer>)
  })

  describe('BodyCopy', () => {
    testRenders(<BodyCopy>This is some body copy text</BodyCopy>)
  })

  describe('Pica', () => {
    testRenders(<Pica>This is some pica text</Pica>)
  })

  describe('PicaSerif', () => {
    testRenders(<PicaSerif>This is some pica serif text</PicaSerif>)
  })

  describe('LongPrimer', () => {
    testRenders(<LongPrimer>This is some long primer text</LongPrimer>)
  })

  describe('Brevier', () => {
    testRenders(<Brevier>This is some brevier text</Brevier>)
  })

  describe('Minion', () => {
    testRenders(<Minion>This is some minion text</Minion>)
  })
})
