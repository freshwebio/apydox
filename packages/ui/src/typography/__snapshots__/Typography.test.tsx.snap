// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography BodyCopy should render without any issues 1`] = `
.c0 {
  font-size: 15px;
  font-family: Lato,sans-serif;
  line-height: 20px;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 16px;
    line-height: 22px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 16px;
    line-height: 22px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 18px;
    line-height: 24px;
  }
}

<p
  className="c0"
  fontFamily="Lato, sans-serif"
  fontSize={
    Array [
      3,
      4,
      4,
    ]
  }
>
  This is some body copy text
</p>
`;

exports[`Typography Brevier should render without any issues 1`] = `
.c0 {
  font-size: 14px;
  font-family: Lato,sans-serif;
  line-height: 16px;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 14px;
    line-height: 18px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 13px;
    line-height: 16px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 14px;
    line-height: 18px;
  }
}

<div
  className="c0"
  fontFamily="Lato, sans-serif"
  fontSize={
    Array [
      2,
      2,
      1,
    ]
  }
>
  This is some brevier text
</div>
`;

exports[`Typography Canon should render without any issues 1`] = `
.c0 {
  font-size: 28px;
  line-height: 32px;
  font-family: Roboto Slab,serif;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 32px;
    line-height: 36px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 44px;
    line-height: 48px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 52px;
    line-height: 56px;
  }
}

<h1
  className="c0"
  fontFamily="Roboto Slab, serif"
  fontSize={
    Array [
      11,
      12,
      14,
    ]
  }
>
  This is some canon text
</h1>
`;

exports[`Typography DoublePica should render without any issues 1`] = `
.c0 {
  font-size: 20px;
  font-family: Lato,sans-serif;
  line-height: 24px;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 20px;
    line-height: 24px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 24px;
    line-height: 28px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 26px;
    line-height: 30px;
  }
}

<div
  className="c0"
  fontFamily="Lato, sans-serif"
  fontSize={
    Array [
      6,
      6,
      9,
    ]
  }
  fontWeight={400}
>
  This is some double pica text
</div>
`;

exports[`Typography GreatPrimer should render without any issues 1`] = `
.c0 {
  font-size: 18px;
  font-family: Roboto Slab,serif;
  line-height: 22px;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 18px;
    line-height: 22px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 20px;
    line-height: 24px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 21px;
    line-height: 24px;
  }
}

<h3
  className="c0"
  fontFamily="Roboto Slab, serif"
  fontSize={
    Array [
      5,
      5,
      6,
    ]
  }
  fontWeight={400}
>
  This is some great primer text
</h3>
`;

exports[`Typography LongPrimer should render without any issues 1`] = `
.c0 {
  font-size: 15px;
  font-family: Lato,sans-serif;
  line-height: 18px;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 15px;
    line-height: 18px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 14px;
    line-height: 18px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 15px;
    line-height: 20px;
  }
}

<div
  className="c0"
  fontFamily="Lato, sans-serif"
  fontSize={
    Array [
      3,
      3,
      2,
    ]
  }
>
  This is some long primer text
</div>
`;

exports[`Typography Minion should render without any issues 1`] = `
.c0 {
  font-size: 12px;
  font-family: Lato,sans-serif;
  line-height: 16px;
  text-transform: uppercase;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 12px;
    line-height: 16px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 12px;
    line-height: 16px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 13px;
    line-height: 16px;
  }
}

<div
  className="c0"
  fontFamily="Lato, sans-serif"
  fontSize={
    Array [
      0,
      0,
      0,
    ]
  }
>
  This is some minion text
</div>
`;

exports[`Typography Paragon should render without any issues 1`] = `
.c0 {
  font-size: 20px;
  font-family: Roboto Slab,serif;
  line-height: 24px;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 22px;
    line-height: 26px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 28px;
    line-height: 32px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 28px;
    line-height: 32px;
  }
}

<h2
  className="c0"
  fontFamily="Roboto Slab, serif"
  fontSize={
    Array [
      6,
      8,
      11,
    ]
  }
  fontWeight={500}
>
  This is some paragon text
</h2>
`;

exports[`Typography Pica should render without any issues 1`] = `
.c0 {
  font-size: 15px;
  font-family: Lato,sans-serif;
  line-height: 20px;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 16px;
    line-height: 20px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 16px;
    line-height: 20px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 18px;
    line-height: 22px;
  }
}

<div
  className="c0"
  fontFamily="Lato, sans-serif"
  fontSize={
    Array [
      3,
      4,
      4,
    ]
  }
>
  This is some pica text
</div>
`;

exports[`Typography Trafalgar should render without any issues 1`] = `
.c0 {
  font-size: 20px;
  line-height: 24px;
  font-family: Roboto Slab,serif;
}

@media screen and (min-width:319px) {
  .c0 {
    font-size: 24px;
    line-height: 28px;
  }
}

@media screen and (min-width:599px) {
  .c0 {
    font-size: 32px;
    line-height: 36px;
  }
}

@media screen and (min-width:599px) and (pointer:coarse) {
  .c0 {
    font-size: 36px;
    line-height: 40px;
  }
}

<h1
  className="c0"
  fontFamily="Roboto Slab, serif"
  fontSize={
    Array [
      6,
      9,
      12,
    ]
  }
>
  This is some trafalgar text
</h1>
`;
